go-controller/pkg/ovn/libovsdbops/util.go

```diff
@@ -2,12 +2,394 @@ package libovsdbops
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"math/rand"
+	"reflect"
+	"strings"
+	"time"
 
+	"github.com/mitchellh/copystructure"
 	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
 	"github.com/ovn-org/libovsdb/ovsdb"
+	"k8s.io/klog"
 )
 
+var (
+	UUIDField = "UUID"
+)
+
+type ModelClient struct {
+	client client.Client
+}
+
+func NewModelClient(client client.Client) ModelClient {
+	return ModelClient{client: client}
+}
+
+/*
+ OnReferentialModelMutation is a helper function which assigns or unassigns
+ all UUIDs provided by `items` against the referential (parent) model.
+
+ Note the following:
+
+- items can be a slice or a unique object. The reasons for this are: a slice
+  makes more sense for delete operations where the client needs to bulk delete a
+  bunch of objects at once (this is done in egressip.go / egressfirewall.go when
+  syncing for example). For create operations usually only one object is to be
+  specified, though multiple can.
+
+- it compares existing referential UUIDs to the ones provided through items, as
+  to avoid round-trips to the OVS DB and unnecessary operations
+*/
+func OnReferentialModelMutation(field interface{}, mutator ovsdb.Mutator, items interface{}) []model.Mutation {
```

Comment:
If I undertood the purpose of this correctly, and in case you are not aware, that kind of filtering is also done server side:

```
         If <mutator> is "insert", then each of the values in the set in
         <value> is added to <column> if it is not already present.  The
         required type of <value> is slightly relaxed, in that it may
         have fewer than the minimum number of elements specified by the
         column's type.

         If <mutator> is "delete", then each of the values in the set in
         <value> is removed from <column> if it is present there.  The
         required type is slightly relaxed in that <value> may have more
         or less than the maximum number of elements specified by the
         column's type.
```

Reply:
Yeah, this was made with regards to: `as to avoid round-trips to the OVS DB`. Doing this will reduce some of the load on the NB, since in cases where we don't even need to call it, we won't.